packages/Operators/test/tstConsistentInterpolationOperator.cpp

```diff
@@ -868,5 +868,111 @@ TEUCHOS_UNIT_TEST( ConsistentInterpolationOperator, local_missed_range_test )
 }
 
 //---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( ConsistentInterpolationOperator, keep_range_data_test )
+{
+    using namespace DataTransferKit;
+
+    // Get the communicator.
+    Teuchos::RCP<const Teuchos::Comm<int> > comm =
+	Teuchos::DefaultComm<int>::getComm();
+    int comm_size = comm->getSize();
+    int comm_rank = comm->getRank();
+
+    // DOMAIN SETUP
+    // Don't put domain entities on proc 0.
+    int num_boxes = (comm->getRank() != 0) ? 5 : 0;
+    Teuchos::Array<DataTransferKit::SupportId> box_ids( num_boxes );
+    Teuchos::ArrayRCP<double> box_dofs( num_boxes );
+    Teuchos::Array<Entity> boxes( num_boxes );
+    for ( int i = 0; i < num_boxes; ++i )
+    {
+	box_ids[i] = num_boxes*(comm_size-comm_rank-1) + i;
+	box_dofs[i] = 2.0*box_ids[i];
+	boxes[i] = BoxGeometry(box_ids[i],comm_rank,box_ids[i],
+		       0.0,0.0,box_ids[i],1.0,1.0,box_ids[i]+1.0);
+    }
+
+    // Make a manager for the domain geometry.
+    DataTransferKit::BasicGeometryManager domain_manager( comm, 3, boxes() );
+    
+    // Make a DOF vector for the domain.
+    Teuchos::RCP<DataTransferKit::Field> domain_field =
+	Teuchos::rcp( new DataTransferKit::EntityCenteredField(
+			  boxes(), 1, box_dofs,
+			  DataTransferKit::EntityCenteredField::BLOCKED) );
+    Teuchos::RCP<Tpetra::MultiVector<double,int,DataTransferKit::SupportId> > domain_dofs =
+	Teuchos::rcp( new DataTransferKit::FieldMultiVector(
+			  domain_field,
+			  domain_manager.functionSpace()->entitySet()) );    
+
+    // RANGE SETUP
+    // Make a range entity set.
+    int num_points = 5;
+    Teuchos::Array<double> point(3);
+    Teuchos::Array<DataTransferKit::SupportId> point_ids( num_points );
+    Teuchos::ArrayRCP<double> point_dofs( num_points );
+    Teuchos::Array<Entity> points( num_points );
+    for ( int i = 0; i < num_points; ++i )
+    {
+	point_ids[i] = num_points*comm_rank + i;
+	point_dofs[i] = 2.2;
+	point[0] = 0.5;
+	point[1] = 0.5;
+	point[2] = point_ids[i] + 0.5;
+	points[i] = Point(point_ids[i],comm_rank,point);
+    }
```

Comment:
5 points per proc, located in the center of boxes of proc+1, except the last proc that will be missed